Source/Core/Request.js

```diff
@@ -0,0 +1,162 @@
+/*global define*/
+define([
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './RequestState',
+        './RequestType'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        RequestState,
+        RequestType) {
+    'use strict';
+
+    /**
+     * Stores information for making a request. This should not be constructed directly. Instead, call the appropriate load function, like `loadWithXhr`, `loadImage`, etc.
+     *
+     * @alias Request
+     * @constructor
+     *
+     * @param {Object} [options] An object with the following properties:
+     * @param {Boolean} [options.url] The url to request.
+     * @param {Function} [options.requestFunction] The actual function that makes the request. The function takes no arguments and returns a promise for the requested data.
+     * @param {Boolean} [options.throttle=false] Whether to throttle and prioritize the request. If false, the request will be sent immediately. If true, the request will be throttled and sent based on priority.
+     * @param {Boolean} [options.throttleByServer=false] Whether to throttle the request by server.
+     * @param {RequestType} [options.type=RequestType.OTHER] The type of request.
+     * @param {Number} [options.distance=0.0] The distance from the camera, used to prioritize requests.
+     * @param {Number} [options.screenSpaceError=0.0] The screen space error, used to prioritize requests.
+     */
+    function Request(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var throttleByServer = defaultValue(options.throttleByServer, false);
+        var throttle = throttleByServer || defaultValue(options.throttle, false);
+
+        /**
+         * The URL to request.
+         *
+         * @type {String}
+         */
+        this.url = options.url;
+
+        /**
+         * The actual function that makes the request. The function takes no arguments and returns a promise for the requested data.
+         *
+         * @type {Function}
+         *
+         * @private
```

Comment:
This and other comments are my overall dilemma with the `Request` class. It needs to be exposed because it is now added to `requestTileGeometry` the `load` functions, but it is not meant to be constructed directly. The one exception might be if someone is calling `requestTileGeometry` but wants to set `throttle : true`. I'm trying to hide the other stuff as much as possible. This is why the documentation also seems conflicting.